build.sh

```diff
@@ -166,6 +166,7 @@ fi
 if hasArg tests || (( ${NUMARGS} == 0 )); then
     BUILD_TESTS=ON
     ENABLE_NN_DEPENDENCIES=ON
+    COMPILE_NN_LIBRARY=ON
```

Comment:
Should we also build the DIST library here? Why do tests only require the NN component?

Reply:
It is a case where none of the tests need any of the dependencies of the distance component. In the case of NN we need the faiss dependency so requesting `COMPILE_NN_LIBRARY` ensures it will be found and linked.